test/helpers/utils.js

```diff
@@ -92,34 +92,24 @@ export function buildSubmitHeaderBlockPaylod(
   if (!wallets) {
     wallets = getWallets()
   }
+
   let validators = options.allValidators
     ? wallets
     : [wallets[1], wallets[2], wallets[3]]
 
-  const extraData = ethUtils.bufferToHex(
-    ethUtils.rlp.encode([
-      [proposer, start, end, root, options.rewardsRootHash, web3.utils.toHex((options.totalStake || '1').toString())] // 0th element
-    ])
-  )
-
-  const vote = ethUtils.bufferToHex(
-    // [chain, voteType, height, round, sha256(extraData)]
-    ethUtils.rlp.encode([
-      'heimdall-P5rXwg',
-      2,
-      0,
-      0,
-      ethUtils.bufferToHex(ethUtils.sha256(extraData))
-    ])
+  let data = web3.eth.abi.encodeParameters(
+    ['address', 'uint256', 'uint256', 'bytes32', 'bytes32', 'bytes32'],
+    [proposer, start, end, root, options.rewardsRootHash, '0x0000000000000000000000000000000000000000000000000000000000003a99']
   )
+  const sigData = Buffer.concat([ethUtils.toBuffer('0x01'), ethUtils.toBuffer(data)])
 
   // in case of TestStakeManger use dummysig data
   const sigs = ethUtils.bufferToHex(
     options.getSigs
-      ? encodeSigs(getSigs(validators, ethUtils.keccak256(vote)))
+      ? encodeSigs(getSigs(validators, ethUtils.keccak256(sigData)))
       : 'dummySig'
   )
-  return { vote, sigs, extraData, root }
+  return { data, sigs, root }
```

Comment:
Is the `root` required anymore?

Reply:
in a few tests we are using is in assertions.

Reply:
This function takes root as a parameter. Please make the assertions on root before calling it. And let's remove it from the return value.

Reply:
Done